client/gutenberg/extensions/simple-payments/edit.js

```diff
@@ -3,34 +3,302 @@
+			content: description,
+			featured_media: 0,
+			meta: {
+				spay_currency: currency,
+				spay_email: email,
+				spay_formatted_price: this.formatPrice( price ),
+				spay_multiple: multiple,
+				spay_price: price,
+			},
+		};
+	};
+
+	savePayment = async () => {
+		if ( this.state.savingProduct ) {
+			return;
+		}
+
+		this.setState( { savingProduct: true } );
+
+		const { attributes, setAttributes } = this.props;
+		const { paymentId } = attributes;
+
+		// @TODO field validation
+
+		const path = `/wp/v2/${ SIMPLE_PAYMENTS_PRODUCT_POST_TYPE }/${ paymentId ? paymentId : '' }`;
+
+		try {
+			// @TODO: then/catch
+			const { id } = await apiFetch( {
+				path,
+				method: 'POST',
+				data: this.attributesToPost( attributes ),
+			} );
+
+			this.setState( { savingProduct: false } );
+
+			if ( id ) {
+				setAttributes( { paymentId: id } );
+			}
+		} catch ( err ) {
+			// @TODO: error handling
+			this.setState( { savingProduct: false } );
+		}
+	};
+
+	// @FIXME: toFixed should be replaced with proper decimal calculations. See simple-payments/form
+	// const { precision } = getCurrencyDefaults( values.currency );
+	formatPrice = price => {
+		price = parseInt( price, 10 );
+		return ! isNaN( price ) ? '$' + price.toFixed( 2 ) : '';
+	};
+
+	handleEmailChange = event => {
+		this.props.setAttributes( { email: event.target.value } );
+	};
+
+	handleDescriptionChange = event => {
+		this.props.setAttributes( { description: event.target.value } );
+	};
+
+	handlePriceChange = event => {
+		const price = parseInt( event.target.value, 10 );
+		if ( ! isNaN( price ) ) {
+			this.props.setAttributes( {
+				formattedPrice: this.formatPrice( event.target.value ),
+				price,
+			} );
 		} else {
-			this.props.setAttributes( { paymentId: undefined } );
+			this.props.setAttributes( {
+				formattedPrice: '',
+				price: undefined,
+			} );
 		}
 	};
 
+	handleCurrencyChange = event => {
+		this.props.setAttributes( {
+			currency: event.target.value,
+			formattedPrice: this.formatPrice( this.props.attributes.price ),
+		} );
+	};
+
+	handleMultipleChange = event => {
+		this.props.setAttributes( { multiple: event.target.checked ? 1 : 0 } );
+	};
+
+	handleTitleChange = event => {
+		this.props.setAttributes( { title: event.target.value } );
+	};
+
 	render() {
-		const { attributes, instanceId } = this.props;
-		const inputId = `payment-id-${ instanceId }`;
+		const { attributes, instanceId, isSelected } = this.props;
+		const {
+			currency,
+			description,
+			email,
+			formattedPrice,
+			multiple,
+			paymentId,
+			price,
+			title,
+		} = attributes;
+		const baseId = `simplepayments-${ instanceId }`;
+		const currencyId = `${ baseId }__currency`;
+		const descriptionId = `${ baseId }__description`;
+		const emailId = `${ baseId }__email`;
+		const multipleId = `${ baseId }__multiple`;
+		const priceId = `${ baseId }__price`;
+		const titleId = `${ baseId }__title`;
+
+		if ( ! isSelected ) {
+			// @TODO component
+			return (
```

Comment:
I'm expecting this is a WIP at this time? Would be even better if we split it off in a separate component before merging, to make it easier for iteration with less chance for conflicts.

Essentially, the more we componentize here, the better it becomes for iteration and reusability.